### PR TITLE
Obsidian config

### PR DESCRIPTION
**Pull request by flozz**
Obsidian Framework configuration implemented :)

### CODE REVIEW

File: obsidian/src/config.test.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        expect(config.get("@app.testValue")).toEqual(42);
+    });
+
+    test("can set and get a value in sub objects", () => {
+        const config = new Config();
+        config.set("@app.obj1.obj2.testValue", 42);
+        expect(config.get("@app.obj1.obj2.testValue")).toEqual(42);
+    });
+
+    test("returns undefined if the did not exist", () => {
```
Comment (tneullas) [inline obsidian/src/config.test.js]: returns undefined if the "PATH" did not exist

File: obsidian/src/helpers.js [hunk 2]
```diff
@@ -34,6 +34,104 @@ const helpers = {
...
+
+    /**
+     * Merges object2 into object1.
+     *
+     * @param {Object} object1 The destination object (will be modified)
+     * @param {Object} object2 The object to merge.
+     *
+     * @return {Object} object1
+     */
+    mergeDeep(object1, object2) {
```
Comment (tneullas) [inline obsidian/src/helpers.js]: Looks like it does like Object.assign() ?

**Comment by tneullas** [comment]
It is written everywhere // eslint-disable-line no-underscore-dangle
It's pretty heavy and repetitive.
If we want to show that they are private then turn eslintrc "no-underscore-dangle" rules to "allow"
`no-underscore-dangle: "allow"`
Otherwise don't use underscore.

File: doc/howto/config.rst [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+Configuration Topology
+----------------------
+
+Obsidian configuration is splitted in 3 categories:
+
+* ``obsidian``: this is the framework configuration itself,
+* ``app``: this is your application global configuration,
+* ``modules``: module-specific configuration.
+
+When you export the obsidian configuration, you will obtain a object that
```
Comment (ladevieq) [inline doc/howto/config.rst]:
```suggestion
When you export the obsidian configuration, you will obtain an object that
```

**Pull request merged**